code/API_definitions/Discovery/README.md

```diff
@@ -0,0 +1,54 @@
+# Edge Discovery APIs
+
+## Simple Discovery API
+This API allows a client application to discover the closest MEC platform to the UE hosting the client application. 'Closest' means 'shorteset network path' as that will give the shortest propogation distance, which is a major factor in latency.
+
+## MEC Experience Management and Exposure API
+This API allows a developer to:
+- discover available MEC platforms, ranked by proximity to a UE.
+- read the state (availability and capabilities) of an operator's various MEC platforms.
+- register a service profile (a description of the developer's edge service) with the MEC operator
+- register the deployed service endpoints with the MEC operator, which allows the closest service endpoint to be discovered at runtime
+
+The API will also support the following capabilities: 
+- events(such as change of status of a MEC platform or another event which could affect their service)
+- subscription to notification of events.
+
+# Mapping to the list of intents
+
+These APIs fulfil the ['discovery' intents](https://github.com/camaraproject/EdgeCloud/blob/main/documentation/SupportingDocuments/Harmonisation%20of%20APIs/describing%20and%20harmonising%20the%20Edge%20APIs.md)
+
+*Simple Edge Discovery* fulfils a single intent,  "4. I can discover the closest MEC platform to a specific terminal (closest in terms of shortest network path)"
+
+*MEC Exposure and Experience Management* is a more comprehensive discovery API and fulfils the following intents:
+
+### Developer intents
+#### Provisioning intents 
+1.	“I can retrieve a list of the operator’s MECs and their status, ordering the results by location and filtering by status (active/inactive/unknown)”
+2.	"I can discover the capabilities/resources available at an operator’s MEC: CPU, Memory, Storage, GPU"
+3.	"I can discover the geographical regions covered by the operators MECs"
+4.	"I can discover the closest MEC platform to a specific terminal (closest in terms of shortest network path)"
+5.	"I can ask the operator to provision my application server to the optimal MEC for a specific terminal, taking into account connectivity, resources (e.g. vCPU, Memory, network interfaces, storage, GPU) shortest network path, cost, network load, MEC platform load, application privacy considerations etc." 
+  *  "I can ask the operator to provision my application server to all MECs that meet these criteria (note this is not focussing on a specific terminal)" 
+  *  "I can ask the operator to provision my application server to a minimal set of MECs that meet these criteria across a given footprint (note this is not focussing on a specific terminal)" 
```

Comment:
I think these intents related to provisioning should be moved to the Workload Orchestration API, otherwise this Discovery API risks becoming a superset of the Workload Orchestration API. There are quite a few Workload Orchestration APIs around artifacts, files, apps, appinstances, etc that would need to be duplicated here to be able to support these intents. I think it would be better to keep the APIs as independent as possible.

Reply:
@maheshc01 please can you advise on @gainsley Jon's question above?

Reply:
As commented during calls I think it make sense to move intent 5 to workload orchestration, as for intent 24 it can stay in discovery although it might be considered duplicated with other intents already included.